src/Enhavo/Bundle/DashboardBundle/Tests/Widget/Type/NumberWidgetTypeTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+
+namespace Enhavo\Bundle\DashboardBundle\Tests\Widget\Type;
+
+
+use Enhavo\Bundle\AppBundle\View\ViewData;
+use Enhavo\Bundle\DashboardBundle\Provider\Provider;
+use Enhavo\Bundle\DashboardBundle\Widget\Type\NumberWidgetType;
+use Enhavo\Component\Type\FactoryInterface;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\OptionsResolver\Exception\MissingOptionsException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class NumberWidgetTypeTest extends TestCase
+{
+    public function testCreateViewData()
+    {
+        $dependencies = $this->createDependencies();
+        $dependencies->factory->method('create')->willReturnCallback(
```

Comment:
Why you use here a callback? Just create the providerMock before and return it with `willReturn`. Callbacks make your code more hard to understand